server/weaverbird/steps/ifthenelse.py

```diff
@@ -0,0 +1,32 @@
+from typing import Any
+
+import numpy
+from pandas import DataFrame
+from pydantic import Field
+
+from weaverbird.steps import BaseStep
+from weaverbird.steps.filter import Condition
+from weaverbird.types import ColumnName, DomainRetriever, PipelineExecutor
+
+
+class IfthenelseStep(BaseStep):
+    new_column: ColumnName = Field(alias='newColumn')
+    condition: Condition = Field(alias='if')
+    then: Any
+    else_value: Any = Field(alias='else')
+
+    def execute(
+        self,
+        df: DataFrame,
+        domain_retriever: DomainRetriever = None,
+        execute_pipeline: PipelineExecutor = None,
+    ) -> DataFrame:
+        # first, I make a series with the True / False result of the condition
+
+        return df.assign(
+            **{
+                self.new_column: numpy.where(
+                    self.condition.filter(df), df.eval(self.then), df.eval(self.else_value)
+                )
+            }
+        )
```

Comment:
This computes the then and else formula for the whole dataframe. It would be faster to just eval the good side of the if/else for each row, but is it doable with pandas ?

Reply:
I think that numpy will apply each evaluated expression only to its related filtered vector so it's just fine in my opinion

Reply:
I tried to apply each eval only to a sub dataframe. Results were not conclusive. cf. https://gist.github.com/luc-leonard/74963aaa2886443bfdd60595e6d8d805